src/Fetch.js

```diff
@@ -150,7 +150,7 @@ function fetchLoadCachedData(db, fetch, onsuccess, onerror) {
         var value = event.target.result;
         var len = value.byteLength || value.length;
 #if FETCH_DEBUG
-        console.log('fetch: Loaded file ' + pathStr + ' from IndexedDB, length: ' + len);
```

Comment:
all these console.logs now became console.errors? Was that an intentional change?

Reply:
What I'm hoping to do is to make all messages that are guarded by XXX_DEBUG go through the same mechanism.   After this change all debug message go to `err`, which is currently mapped to `stderr` on node and `console.warning` on the web.

In the past I think you objects to `err` going to `console.warning` on the web (because it logical should go to `consol.error`.. and I agree with the objection. 

So, as a followup to this change I think we should have a new `log` or `dbg` symbol which exists only in debug builds.  All the XX_DEBUG logging would go through that mechanism and then developers could choose to pipe it to `console.error` or `console.log` as they prefer.   Personally, I'd rather have logging not generate "errors" unless something actually goes wrong, but if its debug-only I think making it configurable is easy and cost-free.

One nice thing about doing that is that one can easily distinguish (using a simple git grep) between actually errors (i.e. something when wrong) and logging/debugging message 

I'll create a followup PR do that now.

Reply:
Yeah, agree in general.

>  Personally, I'd rather have logging not generate "errors"

The main issue I had here is that web browsers do not capture call stacks for console.log and console.warn, only for console.errors. That is why printing all errors to console.warn will impair debuggability, as one will need to then modify the build to be able to see the call stacks of where the messages came from.

(the color yellow/red aspect is not that important (for me at least))

Reply:
Right.. thats why I think we should make logging separate from err, and potentially configurable.  Some folks might want a stack trace for every log line, but I personally like the more traditional printf-style logging. 

I've got a PR in the works that I think will work for everyone.

Reply:
https://github.com/emscripten-core/emscripten/pull/18017